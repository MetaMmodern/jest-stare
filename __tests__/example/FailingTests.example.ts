describe("Failing Tests", () => {

    describe("Snapshot differences", () => {
        it("should fail with a small snapshot difference", () => {
            const jsonObject: any = {
                name: "ein",
                breed: "corgi",
                height: 1,
                color: "brown and white"
            };
            expect(JSON.stringify(jsonObject, null, 2)).toMatchSnapshot();
        });

        it("should fail with a large snapshot difference", () => {
            const jsonObject: any = {
                a: "aardvark",
                b: "balogna",
                c: "cat",
                d: "dog",
                e: "egg",
                f: "frog",
                g: "gecko",
                h: "hyena",
                i: "ice cream",
                j: "jackal",
                k: "koala",
                l: "lemon",
                m: "magpie",
                n: "newt",
                o: "oolong tea",
                p: "parrot",
                q: "quack"
            };
            expect(JSON.stringify(jsonObject, null, 2)).toMatchSnapshot();
        });

        it("should fail with an error snapshot difference", ()=>{
            expect(()=>{
                throw new Error("Here is the new error.\nIt contains various new "+
                "information such as how to write on a computer\n," +
                "and a reminder to scrape an avocado onto your toast.");
            }).toThrowErrorMatchingSnapshot();
        });

    it("should fail due to an error", () => {
        throw new Error("This is an error failure");
    });

    it("Should fail due to a bad expectation", () => {
        expect(false).toEqual(true);
    })
    });
});
