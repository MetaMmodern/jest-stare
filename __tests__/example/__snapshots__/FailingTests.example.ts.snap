// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Failing Tests Snapshot differences should fail with a large snapshot difference 1`] = `
"{
  \\"a\\": \\"aardvark\\",
  \\"b\\": \\"bat\\",
  \\"c\\": \\"cat\\",
  \\"d\\": \\"dog\\",
  \\"e\\": \\"elephant\\",
  \\"f\\": \\"frog\\",
  \\"g\\": \\"gecko\\",
  \\"h\\": \\"hyena\\",
  \\"i\\": \\"iguana\\",
  \\"j\\": \\"jackal\\",
  \\"k\\": \\"koala\\",
  \\"l\\": \\"llama\\",
  \\"m\\": \\"magpie\\",
  \\"n\\": \\"newt\\",
  \\"o\\": \\"ocelot\\",
  \\"p\\": \\"parrot\\"
}"
`;

exports[`Failing Tests Snapshot differences should fail with a small snapshot difference 1`] = `
"{
  \\"name\\": \\"floppy\\",
  \\"breed\\": \\"beagle\\",
  \\"height\\": 1,
  \\"color\\": \\"brown and white\\"
}"
`;

exports[`Failing Tests Snapshot differences should fail with a snapshot difference 1`] = `
"{
  \\"name\\": \\"floppy\\",
  \\"breed\\": \\"beagle\\",
  \\"height\\": 1,
  \\"color\\": \\"brown and yellow\\"
}"
`;

exports[`Failing Tests Snapshot differences should fail with an error snapshot difference 1`] = `
"Here is the old error.
It contains various old information such as how to write on paper
,and a reminder to rewind your VHS tapes before returning them."
`;
